From 28a511b8bf81c0e281dcfb933e5c48aaa65d43ae Mon Sep 17 00:00:00 2001
From: haozho1x <haox.zhou@intel.com>
Date: Fri, 31 Dec 2021 14:27:33 +0800
Subject: [PATCH] Added function of switch display resolution or refresh rateduring Celadon running. 

After connecting through the ADB,change the display mode in the ADB.

Tracked-On: OAM-100301 
Signed-off-by: haozho1x <haox.zhou@intel.com>

---
 drm/DrmConnector.cpp | 4 +++-
 drm/DrmDevice.cpp    | 2 +-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/drm/DrmConnector.cpp b/drm/DrmConnector.cpp
index 3b0f8cb..3e79e03 100644
--- a/drm/DrmConnector.cpp
+++ b/drm/DrmConnector.cpp
@@ -173,10 +173,12 @@ int DrmConnector::UpdateModes() {
 
   int32_t connector_id;
   int32_t mode_id;
+  char ConnectorDynamicId[10]={0};
+  sprintf(ConnectorDynamicId,"%d",id_);
 
   char property[PROPERTY_VALUE_MAX];
   memset(property, 0 , PROPERTY_VALUE_MAX);
-  property_get("vendor.hwcomposer.connector.id", property, "-1");
+  property_get("vendor.hwcomposer.connector.id", property, ConnectorDynamicId);
   connector_id = atoi(property);
   ALOGD("The property 'vendor.hwcomposer.connector.id' value is %d", connector_id);
 
diff --git a/drm/DrmDevice.cpp b/drm/DrmDevice.cpp
index f47aa72..c8d2a77 100644
--- a/drm/DrmDevice.cpp
+++ b/drm/DrmDevice.cpp
@@ -169,7 +169,7 @@ std::tuple<int, int> DrmDevice::Init(const char *path, int num_displays) {
 
   char property[PROPERTY_VALUE_MAX];
   memset(property, 0 , PROPERTY_VALUE_MAX);
-  property_get("vendor.hwcomposer.preferred.mode.limit", property, "1");
+  property_get("vendor.hwcomposer.preferred.mode.limit", property, "0");
   preferred_mode_limit_ = atoi(property) != 0 ? true : false;
   ALOGD("The property 'vendor.hwcomposer.preferred.mode.limit' value is %d", preferred_mode_limit_);
 
-- 
2.17.1

